### ISSUE TITLE
Out of memory reading int16 array

### ISSUE THREAD

**Issue by AgilentGCMS** [Unidata/netcdf4-python#859]
I have a [netcdf file](https://drive.google.com/file/d/1y3KOlaOJj2w5d7p7U89K5MDOOlcm9383/view?usp=sharing) with the following variables:
```
dimensions:
        time_components = 6 ;
...
...
    dimensions:
        n_obs = UNLIMITED ; // (6056468 currently)
    variables:
        double latitude(n_obs) ;
                latitude:unit = "degrees north" ;
        double longitude(n_obs) ;
                longitude:unit = "degrees east" ;
        short times(n_obs, time_components) ;
                times:description = "sounding date and time (UTC)" ;
```
If I open the file and try to read in either latitude or longitude, there is no problem.
```
In [1]: from netCDF4 import Dataset

In [2]: fid = Dataset('decomp.nc', 'r')

In [3]: lons = fid.groups['satellite'].groups['CO'].variables['longitude'][:]
```
The read is instantaneous. However, reading the 'times' array consumes all 32 GB of my RAM and the process is eventually killed by the kernel.
```
In [7]: times = fid.groups['satellite'].groups['CO'].variables['times'][:]
Killed
```
'time_components' is length 6, so the in-memory size of 'lons' should be 47 MB, and that of 'times' should be ~70 MB. Which should not be a problem on my laptop with 32 GB RAM. I suspect there is some memory leak happening somewhere.

My netcdf version is
```
$ nc-config --version
netCDF 4.6.1
$ python -c 'import netCDF4; print netCDF4.__version__'
1.4.2
```

**Comment by jswhit** [comment]
My machine doesn't use a lot of memory reading the variable, but it just takes a *long* time to read it.   An equivalent C program does the same though (a very long time is spent in `nc_get_vara`), so I don't think it's an issue with the python interface.  The reason for this lies in the way the variable is chunked I think:

    from netCDF4 import Dataset
    fid = Dataset('decomp.nc', 'r')
    times = fid.groups['satellite'].groups['CO'].variables['times']
    print times.shape,times.chunking()

    (6056468, 6) [1, 6]


Effectively this means that the reading the entire variable at once will be not much faster than reading one time index at a time (6+ million reads).  If you plan on reading the entire variable at once, I would suggest choosing a different chunksize (other than the default).

**Comment by AgilentGCMS** [comment]
Hmm, on my machine it consumes 32 GB of RAM before being killed. 
Can I set contiguous=True for variables with an unlimited dimension?

**Comment by jswhit** [comment]
No, you can't use contiguous=True with unlimited dimensions.  You can specify a chunksize though.

**Comment by jswhit** [comment]
I am concerned about the large memory consumption - but unfortunately I have not been able to reproduce that.

**Comment by jswhit** [comment]
Curious what happens if you read the variable like this

    times = fid.groups['satellite'].groups['CO'].variables['times']
    tdata = np.empty(times.shape, times.dtype)
    for nt in range(times.shape[0]):
         tdata[nt] = times[nt]

**Comment by AgilentGCMS** [comment]
Jeff,
Reading in the array row by row like you suggested takes a long time (naturally), but only uses ~300 MB of RAM. I had created the decomp.nc file by extracting some variables from another file with ncks;
```
ncks -O -4 -L 9 -v times,measurement,latitude,longitude,model,sigma_mismatch amazon_iasi_finn/output/2010010100-2012050100/aggregated_obs_prior.nc4 amazon_iasi_finn/output/2010010100-2012050100/decomp.nc
```
What I found out is that 'times' in the original file aggregated_obs_prior.nc4 is a contiguous variable. Yet, ncks, when extracting it, chunks it in rows, for some reason.
